### ISSUE TITLE
aws_codebuild_project: fail with environment_variable whose value is ""

### ISSUE THREAD

**Issue by no2a** [terraform-providers/terraform-provider-aws#11192]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform -v
Terraform v0.12.17
+ provider.aws v2.41.0
```

### Affected Resource(s)

* aws_codebuild_project

### Terraform Configuration Files

```hcl
provider "aws" {
  region = "us-west-2"
}

resource "aws_iam_role" "example" {
  name = "example"

  assume_role_policy = <<EOF
{
  "Version": "2012-10-17",
  "Statement": [
    {
      "Effect": "Allow",
      "Principal": {
        "Service": "codebuild.amazonaws.com"
      },
      "Action": "sts:AssumeRole"
    }
  ]
}
EOF
}

resource "aws_codebuild_project" "example" {
  name         = "test-project"
  service_role = aws_iam_role.example.arn

  artifacts {
    type = "NO_ARTIFACTS"
  }

  environment {
    compute_type = "BUILD_GENERAL1_SMALL"
    image        = "aws/codebuild/standard:1.0"
    type         = "LINUX_CONTAINER"

    environment_variable {
      name  = "VAR1"
      value = ""
    }
  }

  source {
    type            = "GITHUB"
    location        = "https://github.com/mitchellh/packer.git"
    git_clone_depth = 1
  }
}
```

### Expected Behavior

a project created with an environment variable VAR1 = "" 

### Actual Behavior

A plan is generated with no value field, and applying it results in an error.
Like this:
```
$ terraform apply
aws_iam_role.example: Refreshing state... [id=example]

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_codebuild_project.example will be created
  + resource "aws_codebuild_project" "example" {
      + arn            = (known after apply)
      + badge_enabled  = false
      + badge_url      = (known after apply)
      + build_timeout  = 60
      + description    = (known after apply)
      + encryption_key = (known after apply)
      + id             = (known after apply)
      + name           = "test-project"
      + service_role   = "arn:aws:iam::xxx:role/example"

      + artifacts {
          + encryption_disabled    = false
          + override_artifact_name = false
          + type                   = "NO_ARTIFACTS"
        }

      + environment {
          + compute_type                = "BUILD_GENERAL1_SMALL"
          + image                       = "aws/codebuild/standard:1.0"
          + image_pull_credentials_type = "CODEBUILD"
          + privileged_mode             = false
          + type                        = "LINUX_CONTAINER"

          + environment_variable {
              + name = "VAR1"
              + type = "PLAINTEXT"
            }
        }

      + source {
          + git_clone_depth = 1
          + location        = "https://github.com/mitchellh/packer.git"
          + type            = "GITHUB"
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

aws_codebuild_project.example: Creating...

Error: Error creating CodeBuild project: InvalidParameter: 1 validation error(s) found.
- missing required field, CreateProjectInput.Environment.EnvironmentVariables[0].Value.


  on pipeline.tf line 24, in resource "aws_codebuild_project" "example":
  24: resource "aws_codebuild_project" "example" {

```

### Steps to Reproduce

1. `terraform apply`

**Comment by no2a** [comment]
I believe emptiness check for the field should be removed. I built a no-check version and successfully created a project with it.

https://github.com/terraform-providers/terraform-provider-aws/blob/v2.41.0/aws/resource_aws_codebuild_project.go#L828

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.62.0 of the Terraform AWS Provider, likely later today. Thanks to @no2a for the implementation. 👍
